### PR TITLE
feat(replays): Improved Replay Details Header

### LINKED ISSUE #36110: Improvement: Condensed Replay Details Header
### Context
The current page header, used in our replay details view, relies on a lot of vertical space to provide the user context of how they got to this replay. The UX is sound, but the space used is excessive.

### Acceptance Criteria
- The user must be able to understand this specific replay was the one they clicked on from the index view.
- The user must be able to navigate back to the replay index page.
- The user must be able to see high-level meta of the replay, similar to the index page.
- The metadata signifiers used on both, the index and details need to be consistent.
- From the relay index, the user can directly go to issues tab to see which issue is featured in this replay.
- Relative timestamp abbreviation used for index and details is consistent with sentry issues

### Designs
[Figma Mock](https://www.figma.com/file/3irLdrxfSf6GxtKmMOM5dq/Exploration%3A-Details-v2?node-id=128%3A5828)
Screen Shot 2022-07-12 at 10 13 46 AM

### Related Issues
https://github.com/getsentry/sentry/issues/35548

### PR DESCRIPTION
**Pull request by jesus4497**
Changed styles for the Replay Header so it can be more condensed. Closes #36110 

image

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

**Comment by jesus4497** [comment]
NOTE 🐛 :

The event inside of a replay detail seems to have the wrong platform value rendering the wrong logo of the project, as you can see in the picture, the element of the replays list is showing correctly the "React" icon, but in the details it is only showing the "JavaScript" icon, this is due to the event having "javascript" as a value in the platform prop.



Replay detail: 
image
```js
{
    "id": "4a4320bdacf54e1ba888a2e70243c8f9",
    "groupID": "3306973937",
    "eventID": "4a4320bdacf54e1ba888a2e70243c8f9",
    "projectID": "6301687",
    "size": 2813,
    "entries": [
        {
            "data": {
                "formatted": "sentry-replay"
            },
            "type": "message"
        },
        {
            "data": {
                "method": null,
                "url": "https://sourcemaps.io/report/1657565618614_https%3A%2F%2Fajax.googleapis.com%2Fajax%2Flibs%2Fangularjs%2F1.5.6%2Fangular.min.js",
                "query": [],
                "fragment": null,
                "data": null,
                "headers": [
                    [
                        "Referer",
                        "https://sourcemaps.io/report/1657565599291_https%3A%2F%2Fajax.googleapis.com%2Fajax%2Flibs%2Fangularjs%2F1.5.6%2Fangular.min.js"
                    ],
                    [
                        "User-Agent",
                        "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36"
                    ]
                ],
                "cookies": [],
                "env": null,
                "inferredContentType": null
            },
            "type": "request"
        }
    ],
    "dist": null,
    "message": "sentry-replay",
    "title": "sentry-replay",
    "location": null,
    "user": {
        "id": null,
        "email": null,
        "username": null,
        "ip_address": "76.9.201.6",
        "name": null,
        "data": null
    },
    "contexts": {
        "browser": {
            "name": "Chrome",
            "version": "103.0.0",
            "type": "browser"
        },
        "device": {
            "family": "Mac",
            "model": "Mac",
            "brand": "Apple",
            "type": "device"
        },
        "os": {
            "name": "Mac OS X",
            "version": "10.15.7",
            "type": "os"
        },
        "trace": {
            "trace_id": "0cdc997d53ea4e3782dca271e917c05a",
            "span_id": "bef838432482e2c1",
            "op": "pageload",
            "status": "unknown",
            "tags": {},
            "type": "trace"
        }
    },
    "sdk": {
        "name": "sentry.javascript.react",
        "version": "6.19.3"
    },
    "context": {},
    "packages": {},
    "type": "default",
    "metadata": {
        "title": "sentry-replay"
    },
    "tags": [
        {
            "key": "browser",
            "value": "Chrome 103.0.0"
        },
        {
            "key": "browser.name",
            "value": "Chrome"
        },
        {
            "key": "device",
            "value": "Mac"
        },
        {
            "key": "device.family",
            "value": "Mac"
        },
        {
            "key": "environment",
            "value": "production"
        },
        {
            "key": "level",
            "value": "error"
        },
        {
            "key": "os",
            "value": "Mac OS X 10.15.7"
        },
        {
            "key": "os.name",
            "value": "Mac OS X"
        },
        {
            "key": "release",
            "value": "9b55903c15725e0f0643e0a483fce4d4573c31b2"
        },
        {
            "key": "sequenceId",
            "value": "0"
        },
        {
            "key": "transaction",
            "value": "/report/1657565618614_https%3A%2F%2Fajax.googleapis.com%2Fajax%2Flibs%2Fangularjs%2F1.5.6%2Fangular.min.js"
        },
        {
            "key": "url",
            "value": "https://sourcemaps.io/report/1657565618614_https%3A%2F%2Fajax.googleapis.com%2Fajax%2Flibs%2Fangularjs%2F1.5.6%2Fangular.min.js"
        },
        {
            "key": "user",
            "value": "ip:76.9.201.6",
            "query": "user.ip:\"76.9.201.6\""
        }
    ],
    "platform": "javascript",
    "dateReceived": "2022-07-13T21:00:18.404170Z",
    "errors": [],
    "_meta": {
        "entries": {},
        "message": null,
        "user": null,
        "contexts": null,
        "sdk": null,
        "context": null,
        "packages": null,
        "tags": {}
    },
    "crashFile": null,
    "culprit": "https://sourcemaps.io/report/1657565618614_https%3A%2F%2Fajax.googleapis.com%2Fajax%2Flibs%2Fangularjs%2F1.5.6%2Fangular.min.js",
    "dateCreated": "2022-07-13T21:00:18.292000Z",
    "fingerprints": [
        "c0b7a27a6a9c104af8ee03566480a165"
    ],
    "groupingConfig": {
        "enhancements": "eJybzDRxY3J-bm5-npWRgaGlroGxrpHxBABcYgcZ",
        "id": "newstyle:2019-10-29"
    },
    "projectSlug": "sourcemapsio-replays",
    "startTimestamp": 1657746017.722,
    "endTimestamp": 1657746018.967
}
```



Replay list element: 
image
```js
{
    "team": {
        "id": "1428824",
        "slug": "sentry-emerging-tech",
        "name": "Sentry Emerging Tech"
    },
    "teams": [
        {
            "id": "1428824",
            "slug": "sentry-emerging-tech",
            "name": "Sentry Emerging Tech"
        }
    ],
    "id": "6477043",
    "name": "ben-test",
    "slug": "ben-test",
    "isBookmarked": false,
    "isMember": true,
    "hasAccess": true,
    "dateCreated": "2022-06-06T22:19:57.073176Z",
    "environments": [
        "demo",
        "%REACT_APP_ENVIRONMENT%"
    ],
    "eventProcessing": {
        "symbolicationDegraded": false
    },
    "features": [
        "alert-filters",
        "custom-inbound-filters",
        "data-forwarding",
        "discard-groups",
        "issue-alerts-targeting",
        "minidump",
        "performance-suspect-spans-ingestion",
        "race-free-group-creation",
        "rate-limits",
        "servicehooks",
        "similarity-indexing",
        "similarity-view",
        "releases"
    ],
    "firstEvent": "2022-06-08T01:42:50.096000Z",
    "firstTransactionEvent": false,
    "hasSessions": true,
    "platform": "javascript-react",
    "platforms": [
        "javascript"
    ],
    "latestRelease": {
        "version": "%REACT_APP_RELEASE%"
    },
    "hasUserReports": false
}
```

**Comment by jesus4497** [comment]
One more thing @ryan953 @Jesse-Box , I didn't put the error metric data as a link because I wasn't sure where should it redirect/land on after clicking on it
image

File: static/app/components/replays/keyMetrics.tsx [hunk 1]
```diff
@@ -1,4 +1,3 @@
-import {Fragment} from 'react';
```
Comment (danecando) [inline static/app/components/replays/keyMetrics.tsx:1]: I've noticed while working on this project it seems like in files where nothing is actually being imported from the react package that it will still include `import React from 'react;` - I know some envs require React to be imported in a file no matter what for the JSX or something like that so I'm assuming it has something to do with that.

**Comment by Jesse-Box** [comment]
That's fine @jesus4497 , thanks for bring it up!

**Review by danecando — COMMENTED** [review]
Just the one comment but it LGTM. 👍

File: static/app/components/replays/keyMetrics.tsx [hunk 2]
```diff
@@ -9,38 +9,24 @@ import space from 'sentry/styles/space';
...
 };
 
-const Key = styled('dt')`
-  color: ${p => p.theme.subText};
-  font-size: 14px;
-  font-weight: bold;
-`;
-
-const Value = styled('dt')`
+const Value = styled('div')`
```
Comment (ryan953) [inline static/app/components/replays/keyMetrics.tsx:26]: looks like there's an extra react component which we probably don't need.

- would it work if we remove the existing `KeyMetricData` which isn't doing much
then we could rename `const Value =` to be anything you like, and export it

Comment (jesus4497) [reply static/app/components/replays/keyMetrics.tsx]: I deleted the component with its storybook as I don't see them necessary anymore and moved the styled components into `detailLayout.tsx` file

File: static/app/views/replays/detail/detailLayout.tsx [hunk 3]
```diff
@@ -120,47 +123,76 @@ function EventMetaData({
...
-      <KeyMetricData
-        keyName={t('Timestamp')}
-        value={event ? <TimeSince date={event.dateReceived} /> : <HeaderPlaceholder />}
+      <ProjectBadge
+        project={{
+          slug: event?.projectSlug || '',
+          id: event?.projectID,
+          platform: event?.platform as PlatformKey,
+        }}
+        avatarSize={16}
```
Comment (billyvg) [inline static/app/views/replays/detail/detailLayout.tsx:132]: This should be a separate PR, but you probably need the `useProjects` hook to fetch the full project to use.

File: static/app/views/replays/detail/detailLayout.tsx [hunk 4]
```diff
@@ -79,32 +93,21 @@ const HeaderPlaceholder = styled(function HeaderPlaceholder(
...
 
-function EventHeader({event}: {event: Event | undefined}) {
-  if (!event) {
-    return <HeaderPlaceholder width="500px" height="48px" />;
-  }
-
-  const urlTag = event.tags.find(({key}) => key === 'url');
-  const pathname = getUrlPathname(urlTag?.value ?? '') ?? '';
+const HeaderContent = styled(Layout.HeaderContent)`
+  margin-bottom: ${space(1)};
```
Comment (ryan953) [inline static/app/views/replays/detail/detailLayout.tsx:97]:
```suggestion
  margin-bottom: 0;
```

**Review by ryan953 — COMMENTED** [review]
We need to have this updated more than detailLayout.tsx

File: static/app/views/replays/detail/detailLayout.tsx (hunk 4 above)
Comment (jesus4497) [reply static/app/views/replays/detail/detailLayout.tsx]: Done! Result: 
image

File: static/app/views/replays/detail/detailLayout.tsx [hunk 5]
```diff
@@ -79,32 +93,22 @@ const HeaderPlaceholder = styled(function HeaderPlaceholder(
...
-      }}
-      // this is the subheading for the avatar, so displayEmail in this case is a misnomer
-      displayEmail={pathname}
-    />
-  );
-}
+const SubHeading = styled('div')`
+  font-size: ${p => p.theme.fontSizeMedium};
+  line-height: ${p => p.theme.text.lineHeightBody};
+  color: ${p => p.theme.gray300};
```
Comment (billyvg) [inline static/app/views/replays/detail/detailLayout.tsx:103]: We should use aliases rather than the color directly:

```suggestion
  color: ${p => p.theme.subText};
```

**Comment by ryan953** [comment]
@jesus4497 we're going to need to followup and update https://github.com/getsentry/sentry/blob/master/static/app/views/replays/detail/page.tsx#L24-L31 with the same changes as well

**Review by ryan953 — APPROVED** [review]
Lets get this merged because it's looking good. I like how narrow the header has become.

The followups are:
1. call `useProjects()` hook so we can get the correct data for `<ProjectBadge>`
2. bring the new styles into the header on `/v2/`, which means editing https://github.com/getsentry/sentry/blob/master/static/app/views/replays/detail/page.tsx
